### DISCUSSION TITLE
Is it possible to access breadCrumbs data in my own client module?

### DISCUSSION THREAD

**Discussion by krillboi** [facebook/docusaurus#7070] [Q&A] [answered]
I would like to access the breadcrumbs data in my analytics module. 

Is this possible at the moment?

**Comment by Josh-Cena** [comment, marked as answer]
Hi, the breadcrumbs data is generated from the current sidebar structure, and the sidebars are available as global data. You can use `import globalData from "@generated/globalData"` and filter out the data that you need.

**Reply by krillboi** [reply]
Great, thanks

**Reply by slorber** [reply]
Not 100% correct, global data does not contain sidebar tree (can be seen here: https://docusaurus.io/__docusaurus/debug/globalData)

These hooks can work for you but they are not part of the public API surface so we might refactor them:

```js
import {useDocsSidebar,  useSidebarBreadcrumbs} from '@docusaurus/theme-common';
``` 

Note that they can only be called on docs-related page, not on client modules, because sidebars data is not available globally


---

It is technically possible to make docs sidebar data available globally (in all pages, client modules...) but this has a performance cost and is a bit more complex, see https://github.com/facebook/docusaurus/issues/4138
